src/Behat/Behat/Context/Environment/Handler/ContextEnvironmentHandler.php

```diff
@@ -155,7 +155,7 @@ function ($context) {
      *
      * @param Suite $suite
      *
-     * @return string[]
+     * @return array<string|array>
```

Comment:
Incorrect tag, the array can also be an array of arrays

Reply:
can we describe the shape of this array ?

Reply:
Not really sure how to best describe this shape because the arrays are also arrays of strings or other arrays and so on recursively, not sure how to best express this

Reply:
So the array case is when the user has specified constructor arguments for their context, I think? In which case the shape is entirely under user control (including that they can optionally specify named params, so the keys could be strings or ints).

It looks to me like these examples from the docs:
```yaml
contexts:
  - MyAwesomeContext:
        - http://localhost:8080
        - /var/tmp
  - MyAwesomeContext:
        baseUrl: http://localhost:8080
        tempPath: /var/tmp
```

stay exactly in that structure here:
```
[
  ['MyAwesomeContext' => ['http://localhost:8080', '/var/tmp'],
  ['MyAwesomeContext' => ['tempPath' => '/var/tmp', 'baseUrl' => 'http://localhost:8080'],
]
```

So I think the best we could do would be `list<string|array<class-string<Context>,array>>`? - unless there's a way to mark that the array version will have exactly **one** `class-string<Context>` key